src/main/utility/childpid_watcher.rs

```diff
@@ -266,35 +263,34 @@ impl ChildPidWatcher {
         unsafe { self.fork_watchable_internal(libc::SYS_fork, child_fn) }
     }
 
-    /// Register interest in `pid`, and associate it with `read_fd`.
-    ///
-    /// `read_fd` should be the read end of a pipe, whose write end is owned
-    /// *solely* by `pid`, causing `read_fd` to become invalid when `pid` exits.
-    /// In a multi-threaded program care must be taken to prevent a concurrent
-    /// fork from leaking the write end of the pipe into other children. One way
-    /// to avoid this is to use O_CLOEXEC when creating the pipe, and then unset
-    /// O_CLOEXEC in the child before calling exec.
+    /// Register interest in `pid`.
     ///
-    /// Be sure to close the parent's write-end of the pipe.
+    /// Will succeed even if `pid` is already dead, in which case callbacks
+    /// registered for this `pid` will run immediately.
```

Comment:
nit: maybe reword to not say immediately? To me, a function saying something will run immediately makes me think it will run in that function (before the function returns), but that's not true in this case.

Reply:
Changed to "will immediately be scheduled to run"